server/services/recommendedVersionsService.test.ts

```diff
@@ -0,0 +1,92 @@
+import RecommendedVersionsService from './recommendedVersionsService'
+import type ServiceCatalogueService from './serviceCatalogueService'
+
+describe('RecommendedVersionsService (Strapi only)', () => {
+  beforeEach(() => {
+    ;(RecommendedVersionsService as unknown as { cache: unknown | null }).cache = null
+  })
+
+  function makeSvcWithComponent(component: unknown, ttlMs = -1) {
+    const svc = new RecommendedVersionsService(ttlMs)
+    const getComponentMock = jest.fn().mockResolvedValue(component)
+    const mockCatalogue = {
+      getComponent: getComponentMock,
+    } as unknown as ServiceCatalogueService
+    svc.setServiceCatalogueService(mockCatalogue)
+    return { svc, mockCatalogue, getComponentMock }
+  }
+
+  it('returns versions from Strapi component (preferred keys)', async () => {
```

Comment:
These tests aren't passing in an input and checking the output is correct for the given input?